test/test_kubeclient.rb

```diff
@@ -343,6 +343,43 @@ def test_init_user_and_bearer_token
     end
   end
 
+  def test_bearer_token_and_bearer_token_file
```

Comment:
tests look good!
I think it's worth to add an additional validation on the specific error messages in all the added tests, since all the failures produce exactly the same ArgumentError, to validate which exact message happens in each case.
It can be done by getting the err returned from assert_raises, and doing an assert_match on the err.message object with the relevant expected message.